external-crates/move/crates/move-cli/src/base/test.rs

```diff
@@ -135,6 +136,7 @@ pub fn run_move_unit_tests<CW: Write + Send, TW: Write + Send>(
     compute_coverage: bool,
     compiler_writer: &mut CW,
     test_writer: &mut TW,
+    report_diags: bool,
```

Comment:
This makes me a bit uncomfortable... just in that flags like this lead to a ton of headaches as they spread (I just had to tear out a bunch of them for testing because they were either not set properly everywhere or ignored in spots).

How much work do you think the real fix is?